### PR TITLE
Add the new functionality in More View controller.

### PR DESCRIPTION
**Pull request by baquer**
Added the new functionality of Tell a friend  in MoreViewController.
issue #56 

screen shot 2018-04-29 at 4 05 53 am

### CODE REVIEW

File: FOSSAsia/MoreViewController.swift [hunk 1]
```diff
@@ -51,6 +106,16 @@ class MoreViewController: UITableViewController {
         case 3:
             self.present(self.createSVC(DefaultURLs.fossasiaGoogleGroups), animated: true, completion: nil)
             break
+        case 4:
+            let mailVc = configureMailComposeViewController(recipient: "",
```
Comment (jogendra) [inline FOSSAsia/MoreViewController.swift]: Please add some other methods like message etc. Have a look at WhatsApp **Tell a friend** feature.

**Review by jogendra — CHANGES_REQUESTED** [review]

File: FOSSAsia/MoreViewController.swift (hunk 1 above)
Comment (baquer) [reply FOSSAsia/MoreViewController.swift]: i will update add other methods like message , facebook etc.

**Comment by baquer** [comment]
screen shot 2018-05-12 at 2 57 23 am
screen shot 2018-05-12 at 2 57 27 am
screen shot 2018-05-12 at 2 57 55 am

**Comment by baquer** [comment]
@mariobehling , @imjog review the PR.

**Review by chashmeetsingh — COMMENTED** [review]
Why not use `UIActivityViewController` to share data with all the applications which can support sharing instead of only having 3 options?

**Comment by baquer** [comment]
@chashmeetsingh, @imjog proposed to  make this as like in WhatsApp so thats why i don't use UIActivityViewController
img_9326 2

**Review by jogendra — CHANGES_REQUESTED** [review]
@baquer 
- WhatsApp is checking whether user's phone is having Facebook, Twitter etc installed or not and show the sharing options according to the availability of Applications and the `More` action is serving the purpose @chashmeetsingh talking about. The same we want to implement here.
- Replace the icon with some png, the current image is showing white corner when we tap on `Tell a Friend` action.
- Do not open messages with empty message string. Use some placeholder text.

**Comment by baquer** [comment]
ok

**Comment by baquer** [comment]
@imjog to see the Message example in action, we  have to run it on an actual device, because the Simulator is unable to send text messages, so please check it using actual device as i performed it on simulator.

**Review by SIMRAN1 — COMMENTED** [review]
@baquer please resolve the failing test!

**Comment by baquer** [comment]
@imjog  i tried to resolve the conflict but UITest and other stuff are working fine, don't know where this failing test is occurring , can you help me to find out ?

**Review by jogendra — COMMENTED** [review]
@baquer Test is failing because of `UIRefreshControl()` which is using outside the main thread which is not allowed. See how @SIMRAN1 solved it in #70. Also put  `UIRefreshControl()` in `DispatchQueue.main.async`.

**Comment by jogendra** [comment]
@baquer Also please look into the changes I requested in last review:

> - WhatsApp is checking whether user's phone is having Facebook, Twitter etc installed or not and show the sharing options according to the availability of Applications and the `More` action is serving the purpose @chashmeetsingh talking about. The same we want to implement here.
> - Replace the icon with some png, the current image is showing white corner when we tap on `Tell a Friend` action.

**Comment by baquer** [comment]
ok will look into it.

**Comment by baquer** [comment]
@imjog if we use UIActivityViewController then , it will automatically pop up the options which are available for  share , but then the problem arises that , how can we implement the "Subject "and "body" in the mail and messages for sharing ?

**Comment by jogendra** [comment]
@baquer 
- Look at the WhatsApp tell a friend feature, they are doing that in `UIActivityViewController` (More) too. You can do this in `UIActivityViewController` too, [see here](https://stackoverflow.com/questions/41695840/add-subject-on-email-by-share-uiactivityviewcontroller-in-swift).

- Also you can easily check whether particular app is installed on user's iPhone or not.

- I resolved failing tests in #73. Wait till it merge

- Add [WIP] before PR title till the work is finished. eg. _[WIP] Add the new functionality in More View controller._

**Comment by baquer** [comment]
@imjog  thanks.

**Pull request closed without merging**